gcovr/writer/html/__init__.py

```diff
@@ -67,15 +67,20 @@ def __call__(self, *args):
         return self.get(*args)
 
 
-# Loading Jinja and preparing the environmen is fairly costly.
+# Loading Jinja and preparing the environment is fairly costly.
 # Only do this work if templates are actually used.
 # This speeds up text and XML output.
 @Lazy
-def templates():
-    from jinja2 import Environment, PackageLoader
+def templates(options):
+    from jinja2 import Environment, PackageLoader, FileSystemLoader
+
+    loader = PackageLoader("gcovr.writer.html")
+
+    if options.template_dir is not None:
+        loader = FileSystemLoader(options.template_dir)
```

Comment:
Why creating a loader if it's immediately overwritten?
```suggestion
    if options.template_dir is None:
        loader = PackageLoader("gcovr.writer.html")
    else:
        loader = FileSystemLoader(options.template_dir)
```